include/dlaf/communication/kernels/all_reduce.h

```diff
@@ -83,37 +92,35 @@ void scheduleAllReduce(const comm::Executor& ex,
   // TILE_O ---> makeContiguous --+--> CONT_BUF_O --+                              |
   //                              |                                                |
   //                              +----------------------> TILE_O -----------------+-> copyBack
-
-  auto ex_copy = getHpExecutor<Backend::MC>();
-
-  pika::future<ContiguousBufferHolder<const T>> cont_buf_in =
-      dataflow(unwrapping(makeItContiguous_o), tile_in);
-
-  pika::future<ContiguousBufferHolder<T>> cont_buf_out;
-  {
-    auto wrapped = getUnwrapRetValAndArgs(
-        dataflow(ex_copy, unwrapExtendTiles(makeItContiguous_o), std::move(tile_out)));
-    cont_buf_out = std::move(wrapped.first);
-    tile_out = std::move(pika::get<0>(wrapped.second));
-  }
-
-  cont_buf_out = getUnwrapReturnValue(dataflow(ex, unwrapExtendTiles(internal::allReduce_o),
-                                               std::move(pcomm), reduce_op, std::move(cont_buf_in),
-                                               std::move(cont_buf_out), tile_in));
-
-  dataflow(ex_copy, unwrapping(copyBack_o), std::move(cont_buf_out), std::move(tile_out));
+  when_all(keep_future(std::move(tile_in)), std::move(tile_out)) |
+      transfer(getBackendScheduler<Backend::MC>()) |
+      let_value(unwrapping(
+          [pcomm = std::move(pcomm), reduce_op](const matrix::Tile<const T, Device::CPU>& tile_in,
+                                                matrix::Tile<T, Device::CPU>& tile_out) mutable {
+            return whenAllLift(std::move(pcomm), reduce_op, makeItContiguous(tile_in),
+                               makeItContiguous(tile_out)) |
+                   transformMPI(internal::allReduce_o) |
+                   transform(Policy<Backend::MC>(thread_priority::high),
+                             std::bind(copyBack_o, std::placeholders::_1, std::cref(tile_out)));
+          })) |
```

Comment:
To make it a little easier to parse, do you mind refactoring the lambda out?

Reply:
Yeah, that's a good idea and I'll make the change here. Also keep in mind that the hope is that https://github.com/eth-cscs/DLA-Future/issues/485 will hopefully make this significantly easier to parse as well. I just don't want to mix that change into this PR. I'm fully aware that the current changes in this PR are not as readable as they could be...

Reply:
I moved the lambda to a separate variable. However, I couldn't think of a good name for it so it's now called `f` :stuck_out_tongue: If you have suggestions for a better name let me know.